src/components/TypewriterModeToggle.tsx

```diff
@@ -15,35 +17,58 @@ interface TypewriterModeToggleProps {
 
 export function TypewriterModeToggle({ typewriterMode, onToggle }: TypewriterModeToggleProps) {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false)
+  const license = useLicenseStore(state => state.license)
+  const hasLicense = Boolean(license?.status === 'active' || license?.status === 'trialing')
```

Comment:
instead of having this logic inside the component, I would make this a function of the store itself so you can do something like:

```
const { hasLicence } = useLicenseStore()
```
Same goes for the other places that hasLicense is being used